### PR TITLE
Rewrite Keycloak article and include sample.

### LINKED ISSUE #2173: Keycloack code fails
### Type of issue

Code doesn't work

### Description

Current State:

- Package is still in pre-release so copy and pasting NuGet command doesn't work
- It feels backwards to add client code before I setup the server in app host
- The pre-reqs section seem to lead me down the path of installing/setting up Keycloak before I do this doc but that seems like not what we would want people to do.
- If you copy and paste the code as-is and F5 you get a HTTPs config error, when you fix that you get a different error.
- We give 2 different code samples for consuming the connection and they are different without any explanation as to why.

Main Proposed Change:

- Rework to assume starting from default starter template and adding KeyCloak, starting with setting up the server then configuring the clients. Show configuring the KeyCloak realm so I can start from just having Aspire and end up with an app I can run and authenticate to.

### Page URL

https://learn.microsoft.com/en-us/dotnet/aspire/authentication/keycloak-integration?tabs=dotnet-cli

### Content source URL

https://github.com/dotnet/docs-aspire/blob/main/docs/authentication/keycloak-integration.md

### Document Version Independent Id

8191aee1-6f2c-5820-046d-17dc81d997e9

### Article author

@IEvangelist

---
[Associated WorkItem - 351267](https://dev.azure.com/msft-skilling/Content/_workitems/edit/351267)

### LINKED ISSUE #2180: Rewrite of the Keycloak integration article
Related to and when this issue is closed, this fixes #2173. /cc @glennc 

---
[Associated WorkItem - 350911](https://dev.azure.com/msft-skilling/Content/_workitems/edit/350911)

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Rewrite Keycloak article and include sample.

- Fixes #2180
- Fixes #2173

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/authentication/keycloak-integration.md](https://github.com/dotnet/docs-aspire/blob/1fa1d7335d13def82dde1d3a7964bbfecc58cc6b/docs/authentication/keycloak-integration.md) | [.NET Aspire Keycloak integration (Preview)](https://review.learn.microsoft.com/en-us/dotnet/aspire/authentication/keycloak-integration?branch=pr-en-us-2231) |
| [docs/toc.yml](https://github.com/dotnet/docs-aspire/blob/1fa1d7335d13def82dde1d3a7964bbfecc58cc6b/docs/toc.yml) | [docs/toc](https://review.learn.microsoft.com/en-us/dotnet/aspire/toc?branch=pr-en-us-2231) |

### CODE REVIEW

**Review by BillWagner — APPROVED** [review]
This looks great @IEvangelist 

I had a couple small comments, but go ahead and :shipit: when ready.

File: docs/authentication/snippets/AspireApp/AspireApp.Web/Components/Pages/Error.razor [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    For local debugging, enable the <strong>Development</strong> environment by setting the <strong>ASPNETCORE_ENVIRONMENT</strong> environment variable to <strong>Development</strong>
+    and restarting the app.
+</p>
+
+@code{
+    [CascadingParameter]
+    public HttpContext? HttpContext { get; set; }
+
+    private string? requestId;
+    private bool ShowRequestId => !string.IsNullOrEmpty(requestId);
```
[the remark this answers is not in the document]
Comment (IEvangelist) [reply docs/authentication/snippets/AspireApp/AspireApp.Web/Components/Pages/Error.razor:32]: This is part of the template. I always use `IsNullOrWhitespace` but the product team seems set on using `IsNullOrEmpty`.

File: docs/authentication/snippets/AspireApp/AspireApp.Web/AuthorizationHandler.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        HttpRequestMessage request, CancellationToken cancellationToken)
+    {
+        var httpContext = httpContextAccessor.HttpContext ??
+            throw new InvalidOperationException("""
+                No HttpContext available from the IHttpContextAccessor.
+                """);
+
+        var accessToken = await httpContext.GetTokenAsync("access_token");
+
+        if (!string.IsNullOrEmpty(accessToken))
```
[the remark this answers is not in the document]
Comment (IEvangelist) [reply docs/authentication/snippets/AspireApp/AspireApp.Web/AuthorizationHandler.cs:19]: I think so. I'll update this, thank you.

Comment (IEvangelist) [inline docs/authentication/snippets/AspireApp/AspireApp.Web/AuthorizationHandler.cs:19]:
```suggestion
        if (!string.IsNullOrWhitespace(accessToken))
```

**Pull request merged**